### DISCUSSION TITLE
Description property for ParseError.Code

### DISCUSSION THREAD

**Discussion by jaysonng** [netreconlab/Parse-Swift#34] [Ideas]
On my project I created an extension to ParseError that basically spits out a description of the Error as so:

```
extension ParseError.Code: CustomStringConvertible {
    public var description: String {
        switch self {
        case .otherCause                      : return "Internal SDK Error. No information available" // used to be .unknownError
        case .internalServer                    : return "Internal server error. No information available."
        case .connectionFailed                  : return "The connection to the Parse servers failed."
        case .objectNotFound                    : return "Object doesn't exist, or has an incorrect password."
        case .invalidQuery                      :
            return "You tried to find values matching a datatype that doesn't support exact database matching, like an array or a dictionary."
        case .invalidClassName                  :
            return "Missing or invalid classname. Classnames are case-sensitive. They must start with a letter, and `a-zA-Z0-9_` are the only valid characters."
        case .missingObjectId                   : return "Missing object id."
        case .invalidKeyName                    :
            return "Invalid key name. Keys are case-sensitive. They must start with a letter, and `a-zA-Z0-9_` are the only valid characters."
            
            ...     
```
            
 The descriptions are basically the comments on each error. 
 
 I was wondering, is this something useful to have as part of the library? 
 
 In my case I use it so that my debugging and error checks are more descriptive. 
 
 I only thought about it now because I just pulled the latest update to ParseSwift and noticed that there were changes to and added errors that I need to update my extension.

**Comment by cbaker6** [comment]
The `ParseError` type has a few extensions to achieve `description`. All Parse types conform to `ParseTypeable`, which requires `CustomStringConvertible`: https://github.com/netreconlab/Parse-Swift/blob/3fd62f4831c0ecd964f1012e5cbf30526707999e/Sources/ParseSwift/Protocols/ParseTypeable.swift#L31-L36

The `codes` you highlighted have the descriptions in their respective documentation. The actual `message` from the error is intended to provide enough detail about what's going on from the server. This allows error messages to only have to be managed from the server level as oppose to managing within the SDK.

**Reply by jaysonng** [reply]
Nice. I think this was added after I made my extension way back when. 

Way cleaner. thanks!
